robocop/checkers/misc.py

```diff
@@ -258,3 +260,25 @@ def visit_LibraryImport(self, node):  # noqa
         if not node.name:
             return
         self.libraries.append(node)
+
+
+class ResourceFileChecker(VisitorChecker):
+    """ Checker for resource files. """
+    rules = {
+        "0912": (
+            "can-be-resource-file",
+            "No tests in '%s' file, consider renaming to '%s.resource'",
+            RuleSeverity.INFO
+        )
+    }
+
+    def visit_File(self, node):  # noqa
```

Comment:
There could be problems with API - it would be good to test if it works with API (for example in intellij plugin)

Reply:
There are, but they appear only when running tests on GitHub, not locally. Do you have an idea why, @bhirsz?

Reply:
Paths are resolved differently:
```
E         "No tests in 'file.robot' file, consider renaming to 'file.resource'"
E         Extra items in the right set:
E         "No tests in 'C:\\directory\\file.robot' file, consider renaming to 'C:\\directory\\file.resource'"
```
It most likely occurs because we're passing Windows-like path in this line: https://github.com/MarketSquare/robotframework-robocop/blob/master/tests/utest/test_api.py#L29 (it would be best to construct Path() and then convert it to str - the slashes will be converted to be compatible with running OS).